pkg/apis/kops/validation/aws.go

```diff
@@ -416,39 +404,39 @@ func hasAWSEBSCSIDriver(c kops.ClusterSpec) bool {
 	return *c.CloudProvider.AWS.EBSCSIDriver.Enabled
 }
 
-func awsValidateAdditionalRoutes(fieldPath *field.Path, routes []kops.RouteSpec, cidr string) field.ErrorList {
+func awsValidateAdditionalRoutes(fieldPath *field.Path, routes []kops.RouteSpec, networkCIDRs []*net.IPNet) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	_, clusterNet, errClusterNet := net.ParseCIDR(cidr)
-	if errClusterNet != nil {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "networking", "networkCIDR"), cidr, "Invalid cluster cidr"))
-	} else {
-		for i, r := range routes {
-			f := fieldPath.Index(i)
-
-			// Check if target is a known type
-			if !strings.HasPrefix(r.Target, "pcx-") &&
-				!strings.HasPrefix(r.Target, "i-") &&
-				!strings.HasPrefix(r.Target, "nat-") &&
-				!strings.HasPrefix(r.Target, "tgw-") &&
-				!strings.HasPrefix(r.Target, "igw-") &&
-				!strings.HasPrefix(r.Target, "eigw-") {
-				allErrs = append(allErrs, field.Invalid(f.Child("target"), r, "unknown target type for route"))
-			}
+	for i, r := range routes {
+		f := fieldPath.Index(i)
+
+		// Check if target is a known type
+		if !strings.HasPrefix(r.Target, "pcx-") &&
+			!strings.HasPrefix(r.Target, "i-") &&
+			!strings.HasPrefix(r.Target, "nat-") &&
+			!strings.HasPrefix(r.Target, "tgw-") &&
+			!strings.HasPrefix(r.Target, "igw-") &&
+			!strings.HasPrefix(r.Target, "eigw-") {
+			allErrs = append(allErrs, field.Invalid(f.Child("target"), r, "unknown target type for route"))
+		}
 
-			ipRoute, _, e := net.ParseCIDR(r.CIDR)
-			if e != nil {
-				allErrs = append(allErrs, field.Invalid(f.Child("cidr"), r, "invalid cidr"))
-			} else if clusterNet.Contains(ipRoute) && strings.HasPrefix(r.Target, "pcx-") {
-				allErrs = append(allErrs, field.Forbidden(f.Child("target"), "target is more specific than cluster CIDR block. This route can target only an interface or an instance."))
+		var routeCIDRs []*net.IPNet
+		allErrs = append(allErrs, validateCIDR(r.CIDR, f.Child("cidr"), &routeCIDRs)...)
```

Comment:
Pre-existing, but I'm not the biggest fan of the validateCIDR method signature, it's pretty hard to grok what's happening there.... might send a follow on PR to clean that up.